source/plugins/ruby/kubernetes_container_inventory.rb

```diff
@@ -107,6 +84,51 @@ def getContainerInventoryRecords(podItem, batchTime, clusterCollectEnvironmentVa
             end
 
             containerInfoMap = containersInfoMap[containerName]
+            # image can be in any one of below format in spec 
+            # repository/image[:imagetag | @digest], repository/image:imagetag@digest, repo/image, image:imagetag, image@digest, image                       
+            imageValue = containerInfoMap["image"]
+            if !imageValue.nil? && !imageValue.empty?
+              # Find delimiters in image format
+              atLocation = imageValue.index("@")
+              isDigestSpecified = false 
+              if !atLocation.nil?
+                # repository/image@digest or repository/image:imagetag@digest, image@digest
+                imageValue = imageValue[0..(atLocation - 1)]
+                # Use Digest from the spec's image in case when the status doesnt get populated i.e. container in pending or image pull back etc.
+                if containerInventoryRecord["ImageId"].nil? || containerInventoryRecord["ImageId"].empty?
+                   containerInventoryRecord["ImageId"] = imageValue[(atLocation + 1)..-1] 
+                end
+                isDigestSpecified = true
+              end
+              slashLocation = imageValue.index("/")
+              colonLocation = imageValue.index(":")
+              if !colonLocation.nil?
+                if slashLocation.nil?
+                  # image:imagetag
+                  containerInventoryRecord["Image"] = imageValue[0..(colonLocation - 1)]                 
+                else
+                  # repository/image:imagetag
+                  containerInventoryRecord["Repository"] = imageValue[0..(slashLocation - 1)]
+                  containerInventoryRecord["Image"] = imageValue[(slashLocation + 1)..(colonLocation - 1)]
+                end
+                containerInventoryRecord["ImageTag"] = imageValue[(colonLocation + 1)..-1]
+              else 
+                if slashLocation.nil?
+                  # image
+                  containerInventoryRecord["Image"] = imageValue
+                else
+                  # repo/image
+                  containerInventoryRecord["Repository"] = imageValue[0..(slashLocation - 1)]
+                  containerInventoryRecord["Image"] = imageValue[(slashLocation + 1)..-1]
+                end 
+                # if no tag specified, k8s assumes latest as imagetag and this is same behavior from docker API and from status.
+                # Ref - https://kubernetes.io/docs/concepts/containers/images/#image-names
+                if isDigestSpecified == false 
+                  containerInventoryRecord["ImageTag"] = "latest"
```

Comment:
Can we remove this? If its not specified, we don't have image tag, rather then filling in with 'latest'.  k8s might assume latest, but we are showing customers what they specified.

Reply:
> Can we remove this? If its not specified, we don't have image tag, rather then filling in with 'latest'. k8s might assume latest, but we are showing customers what they specified.

This is same behavior we have in previous agents (docker & kubelet APIs). removing this would break the customer scenarios if they have any queries or alerts on "latest" image tag.

Reply:
i dont remember seeing hardcoding image to 'latest' anywhere before this change. Where is it hardcoded before to break compatibility ?

Reply:
are you saying previousy we were getting from runtime and it was filling in with latest ? and now we are getting from spec ?

Reply:
Yes, "latest" was coming from the status and same behavior we had with Docker API as well.
For example
# 1. create nginx pod
 kubectl run nginxpod --image=nginx
# 2. observe "latest" tag under status
  containerStatuses:
  - containerID: containerd://93762c31560ecdd8994351d0884b30b8a7d534956c62994b7b44ac38f4266dd1
    image: docker.io/library/nginx:latest